server/build_event_protocol/event_parser/event_parser.go

```diff
@@ -56,8 +57,8 @@ func parseAndFilterCommandLine(in *command_line.CommandLine, allowedEnvVars []st
 	if in == nil {
 		return nil, envVarMap
 	}
-	var out command_line.CommandLine
-	out = *in
+	out := &command_line.CommandLine{}
+	proto.Merge(out, in)
```

Comment:
just looking at this / thinking about it more, another way of fixing this is to change the function signature to:

func parseAndFilterCommandLine(allowedEnvVars []string, in *command_line.CommandLine) map[string]string {
...
}

and just modify in and return the envVarMap

Reply:
I much prefer that solution; if we don't need to preserve the original proto or prevent changes to the original proto from affecting the invocation (which i assume we don't since you're suggesting it), that will be much faster/more memory efficient/more elegant in general.